### PR TITLE
get broadcast to work with YMM registers

### PR DESCRIPTION
**Pull request by WalterBright**
Yay, first executing 32 byte vector code generation!

### CODE REVIEW

File: src/backend/cgxmm.c [hunk 1]
```diff
@@ -1254,7 +1254,9 @@ code *cdvecfill(elem *e, regm_t *pretregs)
         case TYfloat4:
         case TYfloat8:
             if (config.avx &&
-                ((e1->Eoper == OPind && !e1->Ecount) || e1->Eoper == OPvar && !isregvar(e1,&varregm,&varreg)))
+                ((e1->Eoper == OPind && !e1->Ecount) || e1->Eoper == OPvar && !isregvar(e1,&varregm,&varreg)) ||
```
Comment (andralex) [inline src/backend/cgxmm.c]: Can assume "it is known" that && has precedence over ||, so:

```d
e1->Eoper == OPind && !e1->Ecount || e1->Eoper == OPvar && !isregvar(e1,&varregm,&varreg) || tysize(ty) == 32
```

File: src/backend/cgxmm.c [hunk 2]
```diff
@@ -1292,7 +1294,9 @@ code *cdvecfill(elem *e, regm_t *pretregs)
         case TYdouble2:
         case TYdouble4:
             if (config.avx &&
-                ((e1->Eoper == OPind && !e1->Ecount) || e1->Eoper == OPvar && !isregvar(e1,&varregm,&varreg)))
+                ((e1->Eoper == OPind && !e1->Ecount) || e1->Eoper == OPvar && !isregvar(e1,&varregm,&varreg)) ||
+                tysize(ty) == 32
```
Comment (andralex) [inline src/backend/cgxmm.c]: ditto

File: src/backend/cgxmm.c [hunk 3]
```diff
@@ -1343,7 +1347,7 @@ code *cdvecfill(elem *e, regm_t *pretregs)
             cdb.append(c);
             reg -= XMM0;
             cdb.gen2(LODD,modregxrmx(3,reg,r));     // MOVD reg,r
-            checkSetVex(cdb.last(),ty);
+            checkSetVex(cdb.last(),TYschar16);
```
Comment (andralex) [inline src/backend/cgxmm.c]: dem spaces after `,`

**Review by andralex — APPROVED** [review]
Nice Christmas present.  cc @MartinNowak

File: src/backend/cgxmm.c (hunk 1 above)
Comment (WalterBright) [inline src/backend/cgxmm.c]: It is known. Like & over |, and * over +.

Comment (WalterBright) [inline src/backend/cgxmm.c]: Frankly, if someone is so unfamiliar with C, C++, D, what are they doing working on the code generator anyway? :-)

Comment (andralex) [inline src/backend/cgxmm.c]: http://s2.quickmeme.com/img/81/81a797730f10ca7ca92d0ed151bdb690994ccfe2a7247b7afccd6dcc1ed4c446.jpg

**Comment by WalterBright** [comment]
Auto-merge toggled on

File: src/backend/cgxmm.c (hunk 1 above)
Comment (WalterBright) [inline src/backend/cgxmm.c]: http://img13.deviantart.net/a5ad/i/2015/111/a/7/battlestar_galactica__so_say_we_all_by_teighe-d64nzp1.jpg

**Comment by JackStouffer** [comment]
I'm so lucky to be working with people who make battlestar and game of thrones references

File: src/backend/cgxmm.c (hunk 1 above)
Comment (andralex) [inline src/backend/cgxmm.c]: Ehm I noticed the parens remained...

**Pull request merged**